osquery/tables/applications/jetbrains_plugins.cpp

```diff
@@ -0,0 +1,428 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <osquery/core/tables.h>
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/logger/logger.h>
+#include <osquery/tables/applications/jetbrains_plugins.h>
+#include <osquery/tables/system/system_utils.h>
+#include <osquery/utils/conversions/tryto.h>
+#include <osquery/utils/info/platform_type.h>
+#include <regex>
+
+#define LIBARCHIVE_STATIC
+#include <archive.h>
+#include <archive_entry.h>
+
+namespace osquery {
+namespace tables {
+// Function to convert JetBrainsProductType enum to string
+const std::string getProductName(const JetBrainsProductType type) {
+  auto product = kProductTypeToString.find(type);
+  return product->second;
+}
+
+struct FileData {
+  std::string path;
+  std::vector<char> content;
+};
+
+FileData extractSpecificFileFromArchive(const std::string& archive_file_path,
```

Comment:
Wonder if this should be in some common util location, it's a very generic method.